lighthouse-core/lib/network-recorder.js

```diff
@@ -26,10 +27,36 @@ class NetworkRecorder extends EventEmitter {
     this._records = recordArray;
     this.networkManager = NetworkManager.createWithFakeTarget();
 
-    // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
+    this.startedRequestCount = 0;
+    this.finishedRequestCount = 0;
+    this.idleState = true;
+
+    this.networkManager.addEventListener(this.EventTypes.RequestStarted, _ => {
+      this.startedRequestCount++;
+
+      const activeCount = this.startedRequestCount - this.finishedRequestCount;
+      log.verbose('NetworkRecorder', `Request started. ${activeCount} requests in progress` +
+          ` (${this.startedRequestCount} started and ${this.finishedRequestCount} finished).`);
+
+      if (this.idleState) {
+        this.idleState = false;
+        this.emit('networkbusy');
+      }
+    });
+
     this.networkManager.addEventListener(this.EventTypes.RequestFinished, request => {
+      this.finishedRequestCount++;
```

Comment:
> what about failed requests?

as discussed, `RequestFinished` includes failed requests

> and should we handle websocket connections specially or just allow the timeout to address em ?

I actually haven't had any issues with sites that use web sockets so far, weirdly enough. Any good test sites for this?